### ISSUE TITLE
Fail to create Scenario:  KeyError: 'Date'

### ISSUE THREAD

**Issue by hchangac** [lisphilar/covid19-sir#133]
## Summary
Can't create a Scenario when setting Country = 'China', Province = 'Hong Kong'

## Codes and outputs:
```Python
import covsirphy as cs
# Dataset preparation
data_loader = cs.DataLoader("input")
jhu_data = data_loader.jhu()
population_data = data_loader.population()
scenario = cs.Scenario(jhu_data, population_data, country="China", province='Hong Kong')
```
The code returns 
```
KeyError                                  Traceback (most recent call last)
<ipython-input-23-b7feafc92a73> in <module>()
----> 1 scenario = cs.Scenario(jhu_data, population_data, country="China", province='Hong Kong')

7 frames
/usr/local/lib/python3.6/dist-packages/covsirphy/analysis/scenario.py in __init__(self, jhu_data, population_data, country, province)
     40         population_data = self.validate_instance(
     41             population_data, PopulationData, name="population_data")
---> 42         self.population = population_data.value(country, province=province)
     43         # Records
     44         self.jhu_data = self.validate_instance(

/usr/local/lib/python3.6/dist-packages/covsirphy/cleaning/population.py in value(self, country, province)
    144             (int): population in the place
    145         """
--> 146         cell = self.subset(country=country, province=province)
    147         return int(cell.values)
    148 

/usr/local/lib/python3.6/dist-packages/covsirphy/cleaning/cbase.py in subset(self, country, province, start_date, end_date)
    257         if country is None:
    258             ValueError("@country must be specified.")
--> 259         df = self._subset_by_area(country=country, province=province)
    260         # Subset with Start/end date
    261         if start_date is None and end_date is None:

/usr/local/lib/python3.6/dist-packages/covsirphy/cleaning/cbase.py in _subset_by_area(self, country, province)
    230             raise ValueError(
    231                 "@province was specified, but the dataset does not have Province column.")
--> 232         return self._subset_by_province(df, province=province)
    233 
    234     def subset(self, country, province=None, start_date=None, end_date=None):

/usr/local/lib/python3.6/dist-packages/covsirphy/cleaning/cbase.py in _subset_by_province(self, record_df, province)
    191                 raise KeyError(
    192                     f"@province {province} has not been registered.")
--> 193             df = df.groupby(self.DATE).last().reset_index()
    194             return df.drop(self.PROVINCE, axis=1)
    195         # Calculate total values at country level if not registered

/usr/local/lib/python3.6/dist-packages/pandas/core/frame.py in groupby(self, by, axis, level, as_index, sort, group_keys, squeeze, observed)
   5808             group_keys=group_keys,
   5809             squeeze=squeeze,
-> 5810             observed=observed,
   5811         )
   5812 

/usr/local/lib/python3.6/dist-packages/pandas/core/groupby/groupby.py in __init__(self, obj, keys, axis, level, grouper, exclusions, selection, as_index, sort, group_keys, squeeze, observed, mutated)
    408                 sort=sort,
    409                 observed=observed,
--> 410                 mutated=self.mutated,
    411             )
    412 

/usr/local/lib/python3.6/dist-packages/pandas/core/groupby/grouper.py in get_grouper(obj, key, axis, level, sort, observed, mutated, validate)
    598                 in_axis, name, level, gpr = False, None, gpr, None
    599             else:
--> 600                 raise KeyError(gpr)
    601         elif isinstance(gpr, Grouper) and gpr.key is not None:
    602             # Add key to exclusions

KeyError: 'Date'
```
I have tried using Country='China', Province='None', but getting another error
```
ValueError                                Traceback (most recent call last)
<ipython-input-24-b7f99c97e7c2> in <module>()
----> 1 scenario = cs.Scenario(jhu_data, population_data, country="China", province=None)

/usr/local/lib/python3.6/dist-packages/covsirphy/analysis/scenario.py in __init__(self, jhu_data, population_data, country, province)
     50         # First/last date of the area
     51         df = jhu_data.subset(country=country, province=self.province)
---> 52         self.first_date = df[self.DATE].min().strftime(self.DATE_FORMAT)
     53         self.last_date = df[self.DATE].max().strftime(self.DATE_FORMAT)
     54         # Init

pandas/_libs/tslibs/nattype.pyx in pandas._libs.tslibs.nattype._make_error_func.f()

ValueError: NaTType does not support strftime
```

## Environment
- CovsirPhy version: 2.5.4-beta.new.75.fix.130
- Colab

**Comment by hchangac** [comment]
Also not working with "Australia", should I open a new issue? Sorry, I'm still learning how to use GitHub. This is really a fascinating project, thanks for your efforts.

**Comment by lisphilar** [comment]
Dear @hchangac ,
Thank you for using this package and welcome to this project!

Because the root cause seems to be the same for China and Australlia, it is OK to discuss the errors in this issue.

Using the following codes, I found that `jhu_data.subset("China")` and `population_data.subset("Australlia")` returned empty dataframes. This may be the cause of these errors because `Scenario.__init__` calls `subset` methods.

```Python
import covsirphy as cs
data_loader = cs.DataLoader("input")
jhu_data = data_loader.jhu()
population_data = data_loader.population()
# Comfirm that records of China are registered in jhu_data
"China" in jhu_data.countires()
# But empty dataframe was returned for China
jhu_data.subset("China")
# Comfirm that records of Australlia are registered in jhu_data
"Australia" in jhu_data.countires()
# But empty dataframe was returned for Australlia
population_data.subset("Australlia")
```

I will do further investigation to fix this errors.

**Comment by lisphilar** [comment]
Dear @hchangac ,
I tried to fix this issue in development version "2.5.4-delta.new.75.126.fix.130.133" in GitHub.
Please kindly double check it.


1. records in China at country level
I found that the number of recovered cases were zeros in the raw data at country level. I changed the code to replace zeros with actual values. Greater values of [total values of provinces] and [registered values for the country] will be used.

2. records in Australia at country level
Population value was not registered for Australia at country level in the raw data. I changed the codes to calculate total population as the total value of provinces.


3. KeyError for `population_data.value("China", province="Hong Long")` was fixed. (In the previous version, subseted the population data with unregistered column = Date mistakenly.)

Thank you!

**Comment by hchangac** [comment]
yes, it's working pretty well now, again, thanks for your efforts :)
